src/client/__test__/realUserMonitoring.spec.js

```diff
@@ -0,0 +1,155 @@
+// sent by Perfume
+window.requestIdleCallback = jest.fn();
+
+import {realUserMonitoringForPerformance} from '../trackers/realUserMonitoringForPerformance';
+
+describe('src/client/trackers/realUserMonitoringForPerformance', () => {
+
+	describe('.realUserMonitoringForPerformance()', () => {
+
+		beforeEach(() => {
+			realUserMonitoringForPerformance();
+		});
+
+		it('creates a Perfume instance with a custom analytics tracker', () => {
+			expect(Perfume).toHaveBeenCalledTimes(1);
+			const perfumeOptions = Perfume.mock.calls[0][0];
+			expect(typeof perfumeOptions).toBe('object');
+			expect(typeof perfumeOptions.analyticsTracker).toBe('function');
+		});
+
+		describe('analyticsTracker(options)', () => {
+			let analyticsTracker;
+			let oldConsole;
+
+			beforeEach(() => {
+				// We mock the console here so that our library's console.logs don't
+				// make it hard to read the test output
+				oldConsole = global.console;
+				global.console = {log: jest.fn()};
+				analyticsTracker = Perfume.mock.calls[0][0].analyticsTracker;
+			});
+
+			afterEach(() => {
+				global.console = oldConsole;
+			});
+
+			describe('with only one type of metric', () => {
+				beforeEach(() => {
+					analyticsTracker({ metricName: 'fid', data: 13.7 });
+				});
+
+				it('does not broadcast an o-tracking event', () => {
+					expect(broadcast).toHaveBeenCalledTimes(0);
+				});
+			});
+
+			describe('when all metrics are sent', () => {
+				beforeEach(() => {
+					analyticsTracker({ metricName: 'fid', data: 13.7 });
+					analyticsTracker({ metricName: 'lcp', data: 13.7 });
+					analyticsTracker({ metricName: 'ttfb', data: 13.7 });
+					analyticsTracker({ metricName: 'fp', data: 13.7 });
+					analyticsTracker({ metricName: 'fcp', data: 13.7 });
+					analyticsTracker({ metricName: 'cls', data: 13.7 });
+					analyticsTracker({ metricName: 'tbt', data: 13.7 });
+				});
+
+				it('broadcasts an o-tracking event with the formatted metrics data', () => {
+					expect(broadcast).toHaveBeenCalledTimes(1);
+					const broadcastArguments = broadcast.mock.calls[0];
+					expect(broadcastArguments[0]).toBe('oTracking.event');
+					expect(broadcastArguments[1]).toMatchSnapshot();
+				});
+
+				describe('when all metrics are sent a second time', () => {
+					beforeEach(() => {
+						analyticsTracker({ metricName: 'fid', data: 13.7 });
+						analyticsTracker({ metricName: 'lcp', data: 13.7 });
+						analyticsTracker({ metricName: 'ttfb', data: 13.7 });
+						analyticsTracker({ metricName: 'fp', data: 13.7 });
+						analyticsTracker({ metricName: 'fcp', data: 13.7 });
+						analyticsTracker({ metricName: 'cls', data: 13.7 });
+						analyticsTracker({ metricName: 'tbt', data: 13.7 });
+					});
+
+					it('does not broadcast a second o-tracking event', () => {
+						expect(broadcast).toHaveBeenCalledTimes(1);
+					});
+				});
+			});
+
+		});
+
+		describe('when the seed is not in the sample', () => {
+
+			beforeEach(async () => {
```

Comment:
**question:** Is this async needed? I can't see an `await` in the function body.

Reply:
Nope! It's a copy-paste error 🙂 removing